src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -1000,12 +1000,14 @@ bool extend_select_type(const std::shared_ptr<ov::Node>& node, const precisions_
         type_relaxed->set_origin_input_type(ov::element::boolean, 0);
         return true;
     } else if (auto casted = ov::as_type_ptr<ov::op::v1::Select>(node)) {
-        auto relaxed_op =
-            std::make_shared<op::TypeRelaxed<ov::op::v1::Select>>(*casted,
-                                                                  ov::element::TypeVector{ov::element::boolean},
-                                                                  ov::element::TypeVector{});
-        replace_node(node, relaxed_op);
-        return true;
+        if (casted->get_input_source_output(0).get_element_type() != ov::element::boolean) {
```

Comment:
could you add a new test somewhere here
https://github.com/openvinotoolkit/openvino/blob/6f001e9a37d8a3c2264733989f93abd134b58d68/src/common/transformations/tests/utils/convert_precision.cpp#L947 ?